src/System.Private.ServiceModel/tools/scripts/OpenFirewallPorts.cmd

```diff
@@ -5,16 +5,32 @@ REM
 echo Start Open all ports WCF test services use
 REM
 
+SET __EXITCODE=0
+
 netsh advfirewall firewall add rule name="_WCF Test Server PortHttp" dir=in action=allow profile=any localport=8081 protocol=tcp
+call :CheckError %ERRORLEVEL% "_WCF Test Server PortHttp"
+
 netsh advfirewall firewall add rule name="_WCF Test Server PortHttps" dir=in action=allow profile=any localport=44285 protocol=tcp
+call :CheckError %ERRORLEVEL% "_WCF Test Server PortHttps"
+
 netsh advfirewall firewall add rule name="_WCF Test Server PortTcp" dir=in action=allow profile=any localport=809 protocol=tcp
+call :CheckError %ERRORLEVEL% "_WCF Test Server Port
+
 netsh advfirewall firewall add rule name="_WCF Test Server PortWebSocket" dir=in action=allow profile=any localport=8083 protocol=tcp
+call :CheckError %ERRORLEVEL% "_WCF Test Server PortWebSocket"
+
 netsh advfirewall firewall add rule name="_WCF Test Server PortWebSockets" dir=in action=allow profile=any localport=8084 protocol=tcp
+call :CheckError %ERRORLEVEL% "_WCF Test Server PortWebSockets"
 
-REM we do not currently check the error level for the netsh advfirewall command. Will need to investigate the expected error
-REM code in all successful cases and add the check. Will also need to add remove the firewall rules script for clean up
-set __EXITCODE=%ERRORLEVEL%
+goto :done
 
-endlocal
-exit /b %__EXITCODE% 
+:CheckError
+SET _ErrorLevel=%1
+if NOT [%_ErrorLevel%]==[0] if NOT [%_ErrorLevel%]==[1] (
```

Comment:
exit code 1 is OK for removing firewall port, not adding it. I'll remove it here.